src/hotkeys.cpp

```diff
@@ -275,6 +338,28 @@ void Hotkey::AddKeycode(uint16 keycode)
 	include(this->keycodes, keycode);
 }
 
+/**
+ * Appends the hotkeys to the provided string.
+ * @param input The original string to append to.
+ * @param hotkey The hotkey to inspect and append. Can be NULL.
+ */
+std::string AppendHotkeyToString(const std::string& input, const Hotkey* hotkey)
+{
+	if (!hotkey) return input;
+
+	std::stringstream ss;
+	ss << input;
+	if (!hotkey->keycodes.empty()) ss << " ";
+	for (uint16 keycode : hotkey->keycodes)
+	{
+		const bool is_global = keycode & WKC_GLOBAL_HOTKEY;
+		ss << (is_global ? "[" : "(");
+		ss << KeycodeToDisplayString(keycode);
+		ss << (is_global ? "]" : ")");
```

Comment:
Make sure this results in meaningful results on right-to-left languages (mainly Arabic and Hebrew.)